defold-rive/src/comp_rive.cpp

```diff
@@ -1453,17 +1501,24 @@ namespace dmRive
         return false;
     }
 
+    float CompRiveGetDisplayScaleFactor()
+    {
+        return g_DisplayFactor;
+    }
+
     static rive::Vec2D WorldToLocal(RiveComponent* component, float x, float y)
     {
-        float scale = g_DisplayFactor;
-
-        rive::AABB bounds = component->m_ArtboardInstance->bounds();
-        Matrix4 world_inv = dmVMath::Inverse(component->m_World);
-        Vector4 local = (world_inv * Point3(x*scale, y*scale, 0));
-        float bounds_width_half = bounds.width() * 0.5f;
-        float bounds_height_half = bounds.height() * 0.5f;
-        rive::Vec2D p((local.getX() + bounds.width()) / scale, (bounds.height() - local.getY()) / scale);
-        return p;
+        dmGraphics::HContext graphics_context = dmGraphics::GetInstalledContext();
+
+        float window_width = (float) dmGraphics::GetWindowWidth(graphics_context);
+        float window_height = (float) dmGraphics::GetWindowHeight(graphics_context);
+
+        // Width/height are in screen coordinates and not window coordinates (i.e backbuffer size)
+        float normalized_x = x / g_OriginalWindowWidth;
+        float normalized_y = 1 - (y / g_OriginalWindowHeight);
+
+        rive::Vec2D p_local = component->m_InverseRendererTransform * rive::Vec2D(normalized_x * window_width, normalized_y * window_height);
```

Comment:
Save the transform used in the renderer for this artboard (will this actually be a frame late?) and use the inverse to place the x,y in the same space as the artboard. 

There's some hoops here because input.x and input.y are in the projects window coordinates it seems, and we want actual screen coordinates here.